pystac/catalog.py

```diff
@@ -255,10 +255,12 @@ def add_child(
         child.set_root(self.get_root())
         if child.get_parent() is None or not keep_parent:
             child.set_parent(self)
+        elif keep_parent:
+            setattr(child, "keep_parent", True)
```

Comment:
sure yeah if you prefer that it's all the  to me. Do you think we should flip the `keep_parent` kwarg name to match? Maybe `override_parent` so the default is True and then you have to explicitly set it to false?

Reply:
`override_parent` feels strong for what should be a reasonable default thing to do. `set_parent: bool = True` (since that's the method we end up calling)?